crates/agentgateway/src/http/auth.rs

```diff
@@ -9,6 +9,28 @@ use crate::proxy::ProxyError;
 use crate::serdes::deser_key_from_file;
 use crate::*;
 
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
```

Comment:
```suggestion
#[serde(rename_all = "camelCase", deny_unknown_fields, untagged)]
```

Should be a bit more ergonomic, so they don't need to specify the type

Reply:
This ended up on BackendAuth, should be on AwsAuth.

BTW since you initial PR I added `#[apply(schema!)]` which automatically does:

```
#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
#[cfg_attr(feature = "schema", derive(JsonSchema))]
```

So you could do 

```
#[apply(schema!)]
#[serde(untagged)]
```